webui/react/src/components/WorkspaceCreateModal.tsx

```diff
@@ -185,7 +189,10 @@ const WorkspaceCreateModalComponent: React.FC<Props> = ({ onClose, workspaceId }
               required: true,
             },
           ]}>
-          <Input maxLength={80} />
+          <Input
+            disabled={(workspace && !canModifyWorkspace({ workspace })) ?? false}
```

Comment:
is `?? false` necessary? also, do we need these `!canModifyWorkspace` conditionals at all? maybe better to just hide the option to open the modal in the first place if user doesn't have permission (in `WorkspaceActionDropdown`)

Reply:
seems we need the conditional here in order to support "View" mode for this modal. suggest `isViewing` to be more explicit. could also replace conditionals that check `workspaceId` to use `isEditing` instead.

also suggest using more idiomatic `!!workspace` instead of `?? false`

Reply:
Added isViewing and isEditing variables and using them throughout